### ISSUE TITLE
Add (deps (library x)) to depend on an installed library in rules

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3418]
Right now, most of the dune constructs are "referentially transparents", e.g. they don't care if the stanzas are referring to objects installed with opam or are which present in the current workspace. This includes most of the library and executable stanzas (with (libraries ...)`), variables (e.g. %{lib:...}), etc.

However it doesn't seem possible to do the same thing in rules. `(deps (package x))` breaks this: it requires `x` to be present in the workspace and it will fail with `Error: No rule found for alias .x` if `x` is installed with opam. 

I understsand that `dune` doesn't know anything about opam packages, and that requiring an opam package to be installed is a bit weird ; So instead I'm proposing to add a `(deps (library x))` stanza instead, with the same semantics as `(libraries ..)` appearing in libraries/executables. WDYT?

**Comment by jeremiedimino** [comment]
That seems ok at first glance. What about files installed via:

```
(install
 (section lib)
 ...
)
```

Would these be taken into account as well?

**Comment by samoht** [comment]
hum actually using `(package <foo>)` to mean everything installed by `<foo>`, either in the current workspace or by opam would make sense too. The only important thing is to have something which works for both (either changing `(package ...)` or adding `(library ...`)).

**Comment by jeremiedimino** [comment]
Indeed, I would even say that `(package x)` not working when the package is installed is a bug. Everything in dune is supposed to work the same whether dependencies are installed or not

**Comment by ejgallego** [comment]
Very close to a duplicate of #3192
